### PR TITLE
horizon: update changelog

### LINKED ISSUE #887: Update docs to require "horizon serve"
Upcoming release `horizon-v0.17.0` changes the behaviour of the `horizon` binary, so that calling `horizon` with no commands will be an error.

Need to update the repo docs and any guides to say `horizon serve` instead of `horizon` to get the same behaviour.

### PR DESCRIPTION
**Pull request by howardtw**
This PR adds a section for horizon `v0.16.1`

Close #887

### CODE REVIEW

File: services/horizon/CHANGELOG.md [hunk 1]
```diff
@@ -6,6 +6,13 @@ file.  This project adheres to [Semantic Versioning](http://semver.org/).
 As this project is pre 1.0, breaking changes may happen for minor version
 bumps.  A breaking change will get clearly notified in this log.
 
+## v0.16.1
+
+### Changes
+
+* Fixed a bug causing slice bounds out of range at offer-by-account endpoint during streaming.
+* horizon command now requires at least one argument. As a result, users will have to do `horizon serve` to launch horizon.
```
Comment (ire-and-curses) [inline services/horizon/CHANGELOG.md]: Should probably indicate this is a breaking change

File: services/horizon/CHANGELOG.md [hunk 2]
```diff
@@ -6,6 +6,16 @@ file.  This project adheres to [Semantic Versioning](http://semver.org/).
 As this project is pre 1.0, breaking changes may happen for minor version
 bumps.  A breaking change will get clearly notified in this log.
 
+## v0.17.0
+
+### Breaking changes
+
+* horizon command now requires at least one argument. As a result, users will have to do `horizon serve` to launch horizon.
```
Comment (bartekn) [inline services/horizon/CHANGELOG.md]: If it really is a breaking change (and it looks like it is) it should be reverted from master and pushed to `release-horizon-v0.17.0` release branch.

When it comes to header `## v0.17.0` I think it should be `## Unreleased` or something similar. We don't know yet if the next version will be `0.16.1` or `0.17.0`.

**Comment by howardtw** [comment]
@bartekn I just updated the title to `Unrelased`. I haven't merged this one into `master` yet, and I'm not sure how we plan on keeping `0.17.0` synced with `master` during the development process. Changes associated with `cmd` are these: bb67e9b58c600b23b3298db324086f487ceadc38, bf56ad6262e4b0f1ef0c785a9525ec2378c15440, f86cbcc5810bb96a375bd762836537bd87f0ec9e

**Comment by bartekn** [comment]
> I haven't merged this one into `master` yet

Which one do you mean? This one is merged: https://github.com/stellar/go/pull/879.

> I'm not sure how we plan on keeping `0.17.0` synced with `master` during the development process

We will merge master into release branch from time to time or when it's needed.

**Pull request merged**